src/main/kotlin/org/jdc/template/ux/about/AboutViewModel.kt

```diff
@@ -101,6 +107,82 @@ class AboutViewModel
         })
     }
 
+    /**
+     * Individual web service call
+     */
+    fun testIndividualServiceCall() {
+        val call = individualService.individuals()
+
+        call.enqueue(object : Callback<DtoIndividuals> {
+            override fun onResponse(call: Call<DtoIndividuals>, response: Response<DtoIndividuals>) {
+                processWebIndividualServiceResponse(response)
+            }
+
+            override fun onFailure(call: Call<DtoIndividuals>, t: Throwable?) {
+                Timber.e(t, "Individual Search FAILED")
+            }
+        })
+    }
+
+    private fun processWebIndividualServiceResponse(response: Response<DtoIndividuals>) {
+        if (response.isSuccessful) {
+            Timber.i("Individual Search SUCCESS")
+            response.body()?.let {
+                processIndividualSearchResponse(it)
+            }
+        } else {
+            Timber.e("Search FAILURE: code (%d)", response.code())
+        }
+    }
+
+    private fun processIndividualSearchResponse(dtoIndividuals: DtoIndividuals) = launch(cc.commonPool) {
+        val list : MutableList<Individual> = mutableListOf()
+        val result : List<Individual> = list
+
+        val household = Household()
+        household.name = "Shan"
+
+        for (ind in dtoIndividuals.individuals) {
+
+            val thisIndividual = Individual()
+            thisIndividual.id = ind.id
+            thisIndividual.firstName = ind.firstName
+            thisIndividual.lastName = ind.lastName
+            thisIndividual.phone = "801-555-0000"
+            thisIndividual.individualType = IndividualType.HEAD
+            thisIndividual.householdId = household.id
+            val parts = ind.birthdate.split("-")
+            thisIndividual.birthDate = LocalDate.of(parts[0].toInt(), parts[1].toInt(), parts[2].toInt())
+            thisIndividual.alarmTime = LocalTime.of(7, 0)
+            thisIndividual.profileUrl = ind.profilePicture
+            list.add(thisIndividual)
+            Timber.i("Result: %s - %s", thisIndividual.firstName, thisIndividual.profileUrl)
+        }
+        persistIndividuals(result, household)
+    }
+
+    private fun persistIndividuals(individuals : List<Individual>, household: Household) {
+        val individualDao = mainDatabase.individualDao()
+        val householdDao = mainDatabase.householdDao()
+
+        // clear any existing items
+        individualDao.deleteAll()
```

Comment:
Move deleteAll() functions into transaction.... and handle closing transaction in case of a transaction (place in try{ } finally {})

Reply:
If I follow what you are wanting here, I just need to add the try block and a finally to ensure endTransaction is called to either commit or rollback the changes.  There were two places where I noticed this could be done.